tools/checked-c-convert/ProgramInfo.cpp

```diff
@@ -634,72 +757,160 @@ ProgramInfo::getVariableHelper( Expr                            *E,
   }
 }
 
+std::map<std::string, std::set<ConstraintVariable*>>& ProgramInfo::getOnDemandFuncDeclConstraintMap() {
+  return OnDemandFuncDeclConstraint;
+}
+
+std::string ProgramInfo::getUniqueDeclKey(Decl *decl, ASTContext *C) {
+  auto Psl = PersistentSourceLoc::mkPSL(decl, *C);
+  std::string fileName = Psl.getFileName() + ":" + std::to_string(Psl.getLineNo());
+  std::string name = decl->getDeclKindName();
+  if (FunctionDecl *FD = dyn_cast<FunctionDecl>(decl))
+    name = FD->getNameAsString();
+
+  std::string declKey = fileName + ":" + name;
+  return declKey;
+}
+
+std::string ProgramInfo::getUniqueFuncKey(FunctionDecl *funcDecl, ASTContext *C) {
+  // get unique key for a function: which is function name, file and line number
+  if (FunctionDecl *funcDefn = getDefinition(funcDecl))
+    funcDecl = funcDefn;
+
+  return getUniqueDeclKey(funcDecl, C);
+}
+
+std::set<ConstraintVariable*>&
+ProgramInfo::getOnDemandFuncDeclarationConstraint(FunctionDecl *targetFunc, ASTContext *C) {
+  std::string declKey = getUniqueFuncKey(targetFunc, C);
+  if (OnDemandFuncDeclConstraint.find(declKey) == OnDemandFuncDeclConstraint.end()) {
+    const Type *Ty = targetFunc->getTypeSourceInfo()->getTypeLoc().getTypePtr();
+    assert (!(Ty->isPointerType() || Ty->isArrayType()) && "");
+    assert(Ty->isFunctionType() && "");
+    FVConstraint *F = new FVConstraint(targetFunc, freeKey, CS, *C);
+    OnDemandFuncDeclConstraint[declKey].insert(F);
+    // insert into declaration map.
+    CS.getFuncDeclVarMap()[declKey].insert(F);
+  }
+  return OnDemandFuncDeclConstraint[declKey];
+}
+std::set<ConstraintVariable*>
+ProgramInfo::getVariable(clang::Decl *D, clang::ASTContext *C, FunctionDecl *FD, int parameterIndex) {
+  // if this is a parameter.
+  if (parameterIndex >= 0) {
+    // get the parameter index of the
```

Comment:
The wording in the comment isn't accurate here.   `Get the paramater at parameterIndex` would be clearer.

Reply:
Fixed.